### PR TITLE
fix: upgrade lance to 0.7.5 and add tests for searching empty dataset

### PR DESCRIPTION
**Pull request by chebbyChefNEQ**
This PR upgrade lance to `0.7.5`, which include fixes for searching an empty dataset.

This PR also adds two tests in node SDK to make sure searching empty dataset do no throw

### CODE REVIEW

File: node/src/test/test.ts [hunk 1]
```diff
@@ -258,6 +258,34 @@ describe('LanceDB client', function () {
...
+  describe('when searching an empty dataset', function () {
+    it('should not fail', async function () {
+      const dir = await track().mkdir('lancejs')
+      const con = await lancedb.connect(dir)
+
+      const schema = new Schema(
+        [new Field('vector', new FixedSizeList(128, new Field('float32', new Float32())))]
+      )
+      const table = await con.createTable({ name: 'vectors', schema })
+      await table.search(Array(128).fill(0.1)).execute()
```
Comment (changhiskhan) [inline node/src/test/test.ts:270]: can you add an assertion on what it should return?

File: node/src/test/test.ts [hunk 2]
```diff
@@ -258,6 +258,34 @@ describe('LanceDB client', function () {
...
+      const dir = await track().mkdir('lancejs')
+      const con = await lancedb.connect(dir)
+
+      const schema = new Schema(
+        [new Field('vector', new FixedSizeList(128, new Field('float32', new Float32())))]
+      )
+      const table = await con.createTable({ name: 'vectors', schema })
+      await table.add([{ vector: Array(128).fill(0.1) }])
+      await table.delete('vector IS NOT NULL')
+      await table.search(Array(128).fill(0.1)).execute()
```
Comment (changhiskhan) [inline node/src/test/test.ts:285]: same

**Review by changhiskhan — APPROVED** [review]
